src/Data/ConnectionResolver.php

```diff
@@ -298,6 +299,9 @@ public static function get_query_info( $query ) {
 					$query_info['total_items'] = ! empty( $query->get_total() ) ? $query->get_total() : count( $query->get_results() );
 					$query_info['items']       = $query->get_results();
 					break;
+				case is_array( $query ):
```

Comment:
I don't believe this `case is_array( $query ):` will ever evaluate to true as we have a check above: `if ( true === is_object( $query ) ) {`

I think you'd have to approach the same way you did in your other change and have an else statement for arrays.

Reply:
Also, can you elaborate on the use case where this isn't working so I have some context? like maybe share some code or something I can look at?

Reply:
Say this is `get_query` returns an array of arrays
```
protected static function get_query() {
   $query = [
     [ ...itemOne],
     [ ...itemTwo],
     [ ...itemThree],
   ];
   return $query;
}

the current `array_slice` return an empty array. And it called with the `self` keyword in `resolve` so it can't be effectively overloaded. as it is now the only solution is to overload `get_connection` ignore the `$array` and doing reslicing the query results there.